lib/std/mutex.zig

```diff
@@ -26,14 +43,21 @@ pub const Mutex = if (builtin.single_threaded)
             }
         };
 
+        /// Create a new mutex in unlocked state.
         pub fn init() Mutex {
             return Mutex{ .lock = lock_init };
         }
 
+        /// Free a mutex created with init. Calling this while the
+        /// mutex is held may result in safety-checked undefined
```

Comment:
This statement needs to be evaluated for factuality.

Reply:
"Calling this while the mutex is held is illegal behavior."

Not specifying whether illegal behavior is safety checked means it may or may not be checked. Related: #2402

I don't think we can guarantee that it is checked here.